### PR TITLE
Bug fixing for isi function

### PR DESCRIPTION
**Pull request by pietroquaglio**
Two small fixes to the isi function:
* If the input is neo.SpikeTrain wchich t_start is larger than the smallest isi, the function was returning an error, because np.diff(neo.SpikeTrain) returns a neo,SpikeTrain + changed the test to cover such corner case
* Sorted the times of the spikes in input (only if sorted it makes sense to take the diff to compute the isis)

### CODE REVIEW

**Comment by alperyeg** [comment]
I think the sorting issue should also be addressed in the `neo.SpikeTrain` module. What's your thought on this @apdavison ?

**Comment by mdenker** [comment]
I think the multiple if's are redundant, since
`if hasattr(spiketrain, 'waveforms'):`
essentially asks whether this is a spiketrain object. Thus, this if statement should be merged with the preceding
`if isinstance(...`

**Comment by pietroquaglio** [comment]
@mdenker: I agree (done it!)

**Comment by alperyeg** [comment]
I think Michael meant  the other way around. Merge it to the hasattribute condition.

File: elephant/statistics.py [hunk 1]
```diff
@@ -46,8 +46,11 @@ def isi(spiketrain, axis=-1):
...
     if axis is None:
         axis = -1
-    intervals = np.diff(spiketrain, axis=axis)
-    if hasattr(spiketrain, 'waveforms'):
+    if isinstance(spiketrain, neo.SpikeTrain):
+        intervals = np.diff(
+            np.sort(spiketrain.times.view(pq.Quantity)), axis=axis)
+    else:
+        intervals = np.diff(np.sort(spiketrain), axis=axis)
         intervals = pq.Quantity(intervals.magnitude, units=spiketrain.units)
```
Comment (alperyeg) [inline elephant/statistics.py]: This causes an error, since you do not have neo spiketrains anymore but plain numpy arrays.

Comment (pietroquaglio) [inline elephant/statistics.py]: True sorry that was very lazy of me! Going to fix it!
